src/Symfony/Component/Routing/Matcher/StaticUrlMatcher.php

```diff
@@ -0,0 +1,149 @@
+<?php
+
+/*
+ * This file is part of the Symfony package.
+ *
+ * (c) Fabien Potencier <fabien@symfony.com>
+ *
+ * For the full copyright and license information, please view the LICENSE
+ * file that was distributed with this source code.
+ */
+
+namespace Symfony\Component\Routing\Matcher;
+
+use Symfony\Component\Routing\Exception\MethodNotAllowedException;
+use Symfony\Component\Routing\Exception\NoConfigurationException;
+use Symfony\Component\Routing\Exception\ResourceNotFoundException;
+use Symfony\Component\Routing\RequestContext;
+
+/**
+ * StaticUrlMatcher matches URL based on a set of rules dumped by StaticUrlMatcherDumper.
+ *
+ * @author Nicolas Grekas <p@tchwork.com>
+ */
+abstract class StaticUrlMatcher extends UrlMatcher implements RedirectableUrlMatcherInterface
```

Comment:
Are we required to make this an abstract class forcing the child to implement RedirectableUrlMatcherInterface ? The redirectable part is the only un-implemented part.
This PR makes the Routing component unusable in a dumped context without using FrameworkBundle (or implementing your own class).

In the existing matcher, support for redirecting is optional. We could do the same here by removing this abstract interface here (making the class concreate) and adding a `if ($this instanceof RedirectableUrlMatcherInterface)` in places needing to change the logic for redirections ? What would be the performance impact of this check for the framework case where it is redirectable ?